Makefile

```diff
@@ -1,80 +1,95 @@
-
-# Build the image with buildah
-.PHONY: buildah-build
-buildah-build: test
-	BUILDAH_ISOLATION=chroot $(SUDO_CMD) buildah bud --tag ${IMG} .
-
-.PHONY: buildah-push
-buildah-push: buildah-build
-	$(SUDO_CMD) buildah push ${IMG}
+all: build
+.PHONY: all
+
+# Include the library makefile
+include $(addprefix ./vendor/github.com/openshift/build-machinery-go/make/, \
+	golang.mk \
+	targets/openshift/bindata.mk \
+	targets/openshift/crd-schema-gen.mk \
+	targets/openshift/deps.mk \
+)
+
+# adapted from https://github.com/openshift/build-machinery-go/blob/master/make/targets/openshift/images.mk
+
+# IMAGE_BUILD_EXTRA_FLAGS lets you add extra flags for imagebuilder
+# e.g. to mount secrets and repo information into base image like:
+# make images IMAGE_BUILD_EXTRA_FLAGS='-mount ~/projects/origin-repos/4.2/:/etc/yum.repos.d/'
+IMAGE_BUILD_EXTRA_FLAGS ?= --no-cache
+
+# $1 - target name
+# $2 - image ref
+# $3 - Dockerfile path
+# $4 - context
+define build-image-internal
+image-$(1):
+	$(strip \
+		podman build \
+		-t $(2) \
+		-f $(3) \
+		$(IMAGE_BUILD_EXTRA_FLAGS) \
+		$(4) \
+	)
+.PHONY: image-$(1)
+
+images: image-$(1)
+.PHONY: images
+endef
+
+define build-image
+$(eval $(call build-image-internal,$(1),$(2),$(3),$(4)))
+endef
+
+# Set crd-schema-gen variables
+CONTROLLER_GEN_VERSION := v0.2.1
+CRD_APIS :=./pkg/apis/cloudcredential/v1
+
+# Exclude e2e tests from unit testing
+GO_TEST_PACKAGES :=./pkg/... ./cmd/...
+
+IMAGE_REGISTRY :=registry.svc.ci.openshift.org
+
+# This will call a macro called "add-bindata" which will generate bindata specific targets based on the parameters:
+# $0 - macro name
+# $1 - target suffix
+# $2 - input dirs
+# $3 - prefix
+# $4 - pkg
+# $5 - output
+# It will generate targets {update,verify}-bindata-$(1) logically grouping them in unsuffixed versions of these targets
+# and also hooked into {update,verify}-generated for broader integration.
+$(call add-bindata,bootstrap,./bindata/bootstrap/...,bindata,bootstrap,pkg/assets/bootstrap/bindata.go)
+
+# This will call a macro called "build-image" which will generate image specific targets based on the parameters:
+# $0 - macro name
+# $1 - target name
+# $2 - image ref
+# $3 - Dockerfile path
+# $4 - context directory for image build
+$(call build-image,ocp-cloud-credential-operator,$(IMAGE_REGISTRY)/ocp/4.5:cloud-credential-operator, ./Dockerfile,.)
+
+# This will call a macro called "add-crd-gen" will will generate crd manifests based on the parameters:
+# $1 - target name
+# $2 - apis
+# $3 - manifests
+# $4 - output
+$(call add-crd-gen,manifests,$(CRD_APIS),./manifests,./manifests)
+
+update-codegen: update-codegen-crds
+.PHONY: update-codegen
+
+verify-codegen: verify-codegen-crds
+.PHONY: verify-codegen
+
+clean:
+	$(RM) ./cloud-credential-operator
+.PHONY: clean
+
+# TODO targets for backward compatibility while we make the shift in CI
+test-no-gen: test
+.PHONY: test-no-gen
+
+vet: verify-govet
+.PHONY: vet
+
+build-no-gen: build
+.PHONY: build-no-gen
```

Comment:
Is this expected? 

```
❯ make build-no-gen
fatal: No names found, cannot describe anything.
go build -mod=vendor -trimpath -ldflags "-s -w -X github.com/openshift/cloud-credential-operator/pkg/version.versionFromGit="v0.0.0-unknown" -X github.com/openshift/cloud-credential-operator/pkg/version.commitFromGit="342cc25f" -X github.com/openshift/cloud-credential-operator/pkg/version.gitTreeState="clean" -X github.com/openshift/cloud-credential-operator/pkg/version.buildDate="2020-04-08T11:22:36Z" " github.com/openshift/cloud-credential-operator/cmd/cloud-credential-operator
```

Reply:
for `fatal: No names found, cannot describe anything.` yes it is expected.  `git describe` doesn't work if there are no tag, as is the case with this repo.  Probably not forever though.  There has been talk to tag all repos on a per release (or build) basis.